Sources/StreamChat/StateLayer/Coordinators/ChannelListQueryLinkingCoordinator.swift

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Listens to local events and adds or removes channels from ``ChannelListQueryDTO.channels``.
+@available(iOS 13.0, *)
+struct ChannelListQueryLinkingCoordinator {
+    private let eventsController: EventsController
+    private let eventsDelegate: EventsDelegate
+    
+    init(state: ChatListState, query: ChannelListQuery, dynamicFilter: ((ChatChannel) -> Bool)?, channelListUpdater: ChannelListUpdater, eventsController: EventsController, database: DatabaseContainer, clientConfig: ChatClientConfig) {
+        self.eventsController = eventsController
+        eventsDelegate = EventsDelegate(database: database)
+        
+        eventsController.delegate = eventsDelegate
+        eventsDelegate.linkingHandler = { [weak state] channel, shouldLink in
+            guard let state else { return }
+            let channels = await state.value(forKeyPath: \.channels)
+            let isPresent = channels.contains(where: { $0.cid == channel.cid })
+            
+            /// True, if the channel should be part of the current query. Used when handling local changes.
+            let isQueryMatching: (ChatChannel) -> Bool = { channel in
+                if let dynamicFilter {
+                    return dynamicFilter(channel)
+                }
+                if clientConfig.isChannelAutomaticFilteringEnabled {
+                    // When auto-filtering is enabled the channel will appear or not automatically if the
+                    // query matches the DB Predicate. So here we default to saying it always belong to the current query.
+                    return true
+                }
+                return channel.membership != nil
```

Comment:
I guess this was the old fallback. But in reality, this should be false, if automatic filtering is disabled and no filter block is provided, we should not make assumptions, and the channel should not be included ever.
```suggestion
                return false
```